boa/src/syntax/parser/function/mod.rs

```diff
@@ -76,14 +74,19 @@ where
         loop {
             let mut rest_param = false;
 
+            let position = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?.span().start();
             let next_param = match cursor.peek(0)? {
                 Some(tok) if tok.kind() == &TokenKind::Punctuator(Punctuator::Spread) => {
                     rest_param = true;
                     FunctionRestParameter::new(self.allow_yield, self.allow_await).parse(cursor)?
                 }
                 _ => FormalParameter::new(self.allow_yield, self.allow_await).parse(cursor)?,
             };
+            if param_names.contains(next_param.name()) {
+                return Err(ParseError::general("duplicate parameter name", position));
```

Comment:
Could we add the parameter name in the error message? That way it's a bit more explicit.

Reply:
I wanted to, but the `general` function requires a `&'static str`, which I cannot make from a `format!` macro. We would need to change the `ParseError` type to take an `Into<Box<str>>` in order for this to work. I'm not sure if that is within the scope of this PR.

Reply:
Let's leave it like this for now then :)